tools/packages_changes_in_changelog.py

```diff
@@ -0,0 +1,182 @@
+It uses intensively the locally installed git executable capabilities to do so.
+
+The trickiest part of this script is to find the actual commit that needs to be set as the
+base comparison. It is not the commit that holds the version tag. Indeed after a release
+several unrelated changes to an effective modification of the codebase logic are done: this
+concerns in particular all versions values in __init__.py that are updated from 0.XX.dev0 to
+0.XZ once the candidate release branch is merged back to master.
+
+Also, the particular situation of point release branches needs to be taken into account, since
+theses releases are not extracted from master, but from the current 0.XX.x branch, on top of
+the previous point release.
+
+The correct commit base is the commit that integrates the commit tagged with the given base
+release. Indeed, this commit contains all unrelated changes to codebase logic and defines the
+new integration window for the next release: from it the diff contains the effective modifications
+for the next release.
+
+Then the two main use cases for this script are (we take 0.34.1 as the current release version):
+1) For a normal release: after extracting candidate-0.35.0 from master, call
+   `tools/packages_changes_in_changelog.py 0.34.0`
+   (here base comparison is last normal version, so 0.34.0)
+2) For a point release: after checkout of current release branch 0.34.x, call
+   `tools/packages_changes_in_changelog.py 0.34.1`
+   (here base comparison is last point version, so 0.34.1)
+
+WARNING: You need to execute the script BEFORE executing the release script since it will update
+         all versions in __init__.py files. Otherwise the CHANGELOG.md will contain all the
+         distributions since they have all been modified by the release script.
+"""
+import argparse
+import os
+import re
+import subprocess
+
+
+# Define distributions to be ignored in the changelog, most like because they are not released.
+IGNORE_DISTRIBUTIONS = ['certbot-postfix', 'certbot-compatibility-tests', 'certbot-ci']
+
+# For the certbot distribution, gives the relative paths to search for, ignoring all others.
+# These paths can be files or directories.
+CERTBOT_DISTRIBUTION_RELATIVE_PATHS = ['setup.py', 'setup.cfg', 'certbot', 'docs',
+                                       'pytest.ini', 'local-oldest-requirements.txt']
+
+CHANGELOG_HEADER_TPL = '''\
+Despite us having broken lockstep, we are continuing to release new versions of
+all Certbot components during releases for the time being, however, the only
+package with changes other than its version number since {0} was:'''
+
+CHANGELOG_FOOTER = '''\
+More details about these changes can be found on our GitHub repo.'''
+
+
+def _find_certbot_root_path():
+    output = subprocess.check_output(['git', 'rev-parse', '--show-toplevel'])
+    return output.strip()
+
+
+def _find_last_release_merge_commit(base_version):
```

Comment:
I'm not sure this function works. I may be misunderstanding something, but I put the file on the `master` branch (as of 8b684e9b9543c015669844222b8960e1b9a71e97) and called it with `0.34.0` and `0.34.2`. Both times it returned d391fb8876dd97d5f446bcd1667ecdce0c4b187c. Shouldn't calling it with `0.34.0` return an earlier commit?